### ISSUE TITLE
[Help]: Regarding the MaskGCT input params  and the HF demo

### ISSUE THREAD

**Issue by treya-lin** [open-mmlab/Amphion#305]
Hi I tried the hf demo of maskgct and also deployed it on my local server. 

I noticed that on Huggingface demo page, the user only needs to input the prompt audio and target text, but in the example code of README, it requires prompt audio, prompt text, prompt lang, target lang, target text. 

I wonder if the three params (prompt text, prompt lang, target lang) must be given? Or are they optional? 
If they must be given, does the demo did asr and lang id detect at the server end? any chance the code for setting up that demo can be open to the public?

**Comment by HeCheng0625** [comment]
Hi, in the huggingface demo, we use an ASR model directly to detect the prompt text, prompt lang, and target lang. You can follow the code: https://huggingface.co/spaces/amphion/maskgct/blob/main/app.py

**Comment by treya-lin** [comment]
> Hi, in the huggingface demo, we use an ASR model directly to detect the prompt text, prompt lang, and target lang. You can follow the code: https://huggingface.co/spaces/amphion/maskgct/blob/main/app.py

Oh thank you! That's very helpful.

I am also hoping there could be more info about, for instance,  the restrictions of input, like,  
1. how should we clean or normalize the prompt text and target text so that it doesn't contain symbols or characters that are difficult for the model to synthesize? Should there be punctuations or not? 
2. (Most importantly) is there any restriction on the prompt wav? I mean, is it the longer the better? Or too long is not ideal? when prompt audio is quite sufficient, should we restrict it to certain length?

I am also testing around and looking into the codebase , but still wanna ask because I guess you guys already have much more experience with this. thank you for the great work and very looking forward to the finetuning guide.

**Comment by treya-lin** [comment]
> > > Hi, in the huggingface demo, we use an ASR model directly to detect the prompt text, prompt lang, and target lang. You can follow the code: https://huggingface.co/spaces/amphion/maskgct/blob/main/app.py
> > 
> > 
> > Oh thank you! That's very helpful.
> > I am also hoping there could be more info about, for instance, the restrictions of input, like,
> > 
> > 1. how should we clean or normalize the prompt text and target text so that it doesn't contain symbols or characters that are difficult for the model to synthesize? Are punctuations preferred? If the text has no punc would it hurt the result?
> > 2. (Most importantly) is there any restriction on the prompt wav? I mean, is it the longer the better? Or too long is not ideal? when prompt audio is quite sufficient, should we restrict it to certain length?
> > 
> > I am also testing around and looking into the codebase , but still wanna ask because I guess you guys already have much more experience with this. thank you for the great work and very looking forward to the finetuning guide.
> 
> Hi, MaskGCT is trained on Emilia, which means its training data is in 3-30 seconds. So it might be helpful to:
> 
> 1. It is recommended that punctuation for prompt_text be added.
> 2. Cut the prompt_wav into 3-10 seconds. (Make sure the total length of prompt wav and generated wav is within 30 seconds)
> 3. If possible, provide a clean prompt_wav; you may use source separation models like UVR or Mel-Band-Roformer to get a clean vocal.
> 
> (P.S. Just notice that we did not release the duration distribution of Emilia, attached here, for other details, feel free to check [Arxiv](https://arxiv.org/abs/2407.05361)) ![image](https://private-user-images.githubusercontent.com/44130602/387301991-c9d847c8-6b9c-4771-be8d-60e0799b3ddf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5ODIwNTMsIm5iZiI6MTczMTk4MTc1MywicGF0aCI6Ii80NDEzMDYwMi8zODczMDE5OTEtYzlkODQ3YzgtNmI5Yy00NzcxLWJlOGQtNjBlMDc5OWIzZGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE5VDAyMDIzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkNGFkNzgxYmMwMGY2N2FjYzlmODdjNTQ4ODdiOGUwN2VhMTQ4NDA2MDVhNDY5NTI0NGUzNWQ3NGVjNzAxM2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dHglKRmnm9rLQKF2Oa1tDYws3Yd2eFjxXw_QRRr1MB8)

Thank you for the reply. This information is indeed very helpful! This issue has been solved and will be closed. Thanks!
